File: csharp/Betfair.ESAClient/Betfair.ESAClient/Protocol/StatusException.cs
using Betfair.ESASwagger.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Betfair.ESAClient.Protocol
{
    /// <summary>
    /// Exception used by api to raise a status fail.
    /// </summary>
    public class StatusException : Exception
    {
        public readonly StatusMessage.ErrorCodeEnum ErrorCode;
        public readonly string ErrorMessage;

        public StatusException(StatusMessage message) : base(message.ErrorCode +": " +message.ErrorMessage)
        {
            ErrorCode = (StatusMessage.ErrorCodeEnum)message.ErrorCode;
            ErrorMessage = message.ErrorMessage;
        }
    }
}

File: java/client/src/main/java/com/betfair/esa/client/protocol/StatusException.java
package com.betfair.esa.client.protocol;

import com.betfair.esa.swagger.model.StatusMessage;

/** Created by mulveyj on 07/07/2016. */
public class StatusException extends Exception {

    private final StatusMessage.ErrorCodeEnum errorCode;
    private final String errorMessage;

    public StatusException(StatusMessage message) {
        super(message.getErrorCode() + ": " + message.getErrorMessage());
        errorCode = message.getErrorCode();
        errorMessage = message.getErrorMessage();
    }

    public StatusMessage.ErrorCodeEnum getErrorCode() {
        return errorCode;
    }

    public String getErrorMessage() {
        return errorMessage;
    }
}
